pytorch_lightning/plugins/environments/lightning_environment.py

```diff
@@ -46,8 +48,18 @@ def master_port(self) -> int:
             self._master_port = os.environ.get("MASTER_PORT", find_free_network_port())
         return int(self._master_port)
 
-    def world_size(self) -> Optional[int]:
-        return None
+    def world_size(self) -> int:
+        return self._world_size
+
+    def set_world_size(self, size: int) -> None:
```

Comment:
Do we need set_world_size. Can you just use a setter and make everything properties ?

Reply:
Yes but this interface is old and all getters are methods from the beginning. 
It would be better to have proper setters and getters. I decided not to keep backward compatibility and make the interface consistent. I propose to do a deprecation in a follow up to keep this PR managable